### ISSUE TITLE
Add option to only "Spellcheck on Save"

### ISSUE THREAD

**Issue by milesfrain** [streetsidesoftware/vscode-spell-checker#493]
An option to only "Spellcheck on Save" would help reduce CPU usage. My fans spin-up whenever I'm actively typing with this extension enabled. I suspect this extension is re-running spellcheck on the whole file for every new keystroke entered.

CPU usage immediately returns to low levels while idle. So I believe this is expected behavior, especially when working on a 10k word file, rather than a bug (like the runaway regex issue as described in #431 that permanently spiked CPU).

My workaround for now to just try to remember to toggle-on spellcheck before commits, but that's not the most ideal workflow.

**Comment by milesfrain** [comment]
I now realize there might be a dilemma with the "Spellcheck on Save" suggestion:

If you want to be able to verify that flagged typos are being fixed in real-time, the extension needs to be enabled during that process. Otherwise the user would need to re-save to re-check each edit.

Perhaps there's a more efficient way to spellcheck, such as only checking edited content since the last spellcheck, rather than the whole file. Maybe this extension is already taking some of these efficiency steps. I haven't dug into the source of this extension, so forgive my armchair advice.

**Comment by Jason3S** [comment]
Do you have an example that exhibits the issue? A 10K file should not be an issue.

**Comment by milesfrain** [comment]
Here's one of the files: https://github.com/milesfrain/purescript-book/blob/ch10-rewrite/text/chapter10.md

**Comment by Jason3S** [comment]
Nice book.

I have cloned the book and tried editing the files. I don't see any performance issues. Spell checking is happening as I type.

Would you share your `cSpell` settings? I think there might be an issue with one of the ignoreRegExpList items. Are you trying to ignore code blocks?

**Comment by milesfrain** [comment]
User Settings:
```json
    "cSpell.userWords": [
        "adt",
        "adts",
        "arities",
        "effectful",
        "foldr",
        "forall",
        "imag",
        "inlines",
        "lebab",
        "lmap",
        "newtype",
        "newtypes",
        "parametricity",
        "purescript",
        "purs",
        "spago",
        "typechecker",
        "typeof",
        "uncurried"
    ],
    "cSpell.enableFiletypes": [
        "purescript"
    ],
    "cSpell.enabled": false,
```

Workspace settings:
```json
  "cSpell.enabled": true
```

Here are some linux commands to produce a log of CPU usage (2x per second) for this extension. You can also just look at `htop`, but it's tough to copy-paste that. Adopted from https://stackoverflow.com/a/52751050

```bash
CSPELL=$(pgrep -f streetsidesoftware.code-spell-checker)
ps $CSPELL
while true; do top -b -n 1 -d 0.5 -p $CSPELL | tail -1 | awk '{print $9}' | tee -a cspell-log.txt ; done

# To stop script, press `Ctrl + Z`, then
kill %1
```

Here's my output of `cspell-log.txt`. The numbers in the middle are when typing. 

```
0.0
0.0
0.0
6.7
113.3
100.0
93.3
120.0
80.0
120.0
106.7
0.0
0.0
0.0
```

**Comment by Jason3S** [comment]
Thank you for the example.

Did you try increasing the delay?
![image](https://user-images.githubusercontent.com/3740137/82356072-66e3b580-9a03-11ea-9199-ec056cc0b5b4.png)

**Comment by milesfrain** [comment]
Nice! Didn't see that. Might be good to add to the CPU performance issues [checklist](https://github.com/streetsidesoftware/vscode-spell-checker/issues/431#issuecomment-586713840).

This is a major improvement. Still a pretty significant blip on CPU usage whenever it fires, but certainly not as bad as it was before. 

A possible debouncer improvement is to **cancel the delay upon file save**. That way I can keep the interval really high and effectively enable the format-on-save feature.

CPU log for reference.
```
0.0
0.0
0.0
46.7
140.0
86.7
0.0
33.3
0.0
0.0
0.0

```

**Comment by mfp22** [comment]
I just find it extremely distracting while I'm typing. I would like to either have it only check on save, or just use a command to ask it to check the current file once.

**Comment by mfp22** [comment]
Spell Check Delay seems to work great. I set it to 60000 and when I'm away for a minute and won't be distracted by it, it works, or when I save it immediately works. 

One issue I didn't notice until I messed with this is that the squiggle never goes away until I close and reopen the file. I tried all kinds of things 

![Image](https://github.com/user-attachments/assets/775f9daf-9338-43ce-9d0f-022b5cf2780a)

**Comment by Jason3S** [comment]
@mfp22,

How about just hiding the spelling issues?

<img width="613" alt="Image" src="https://github.com/user-attachments/assets/6a2e1d5b-829d-443b-96f1-504fc1cc67c2" />

| Title | Command |
| --- | --- |
| Spell: Toggle Show Spelling Issues | `cSpell.toggleVisible` |
| Spell: Hide Spelling Issues | `cSpell.hide` |
| Spell: Show Spelling Issues | `cSpell.show` |

You can bind them to the keyboard:

<img width="643" alt="Image" src="https://github.com/user-attachments/assets/9cc9ce42-4708-4d32-8bbe-e0ba52c3a5fa" />
